com.abdownloadmanager.abdownloadmanager.yml

```diff
@@ -0,0 +1,46 @@
+app-id: com.abdownloadmanager.abdownloadmanager
+runtime: org.freedesktop.Platform
+runtime-version: "24.08"
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk21
+command: ABDownloadManager
+finish-args:
+  - --share=network
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
+  - --device=dri
+  - --filesystem=xdg-download
+  - --persist=.abdm
+  - --env=PATH=/app/bin:/app/jre/bin:/usr/bin
+  - --env=JAVA_HOME=/app/jre
+
+modules:
+  - name: openjdk
+    buildsystem: simple
+    build-commands:
+      - /usr/lib/sdk/openjdk21/install.sh
+
+  - name: ABDownloadManager
+    buildsystem: simple
+    build-commands:
+      - mkdir /app/bin
+      - install -Dm755 bin/ABDownloadManager /app/bin/
+      - cp -r lib /app/
+      - install -Dm644 -t /app/share/applications $FLATPAK_ID.desktop
+      - install -Dm644 -t /app/share/metainfo $FLATPAK_ID.metainfo.xml
+      - install -Dm644 lib/ABDownloadManager.png /app/share/icons/hicolor/512x512/apps/$FLATPAK_ID.png
+    sources:
+      - type: file
+        path: com.abdownloadmanager.abdownloadmanager.desktop
+      - type: file
+        path: com.abdownloadmanager.abdownloadmanager.metainfo.xml
+      - type: archive
+        url: https://github.com/amir1376/ab-download-manager/releases/download/v1.6.10/ABDownloadManager_1.6.10_linux_x64.tar.gz
```

Comment:
What happen on aarch64?

Reply:
The compiling process of this project relies on the internet connection. It uses https://gradle.org/, and the scripts download the latest version of gradle to start compiling. Even adding the gradle manually to the flatpak, gradle itself still downloads many dependencies afterwards